### PR TITLE
Update form data api defaults

### PR DESCRIPTION
**Pull request by adfoster-r7**
Updates the API To:
- Remove the 8bit defaults from `form_data` as it's not common. The default is assumed to be 7bit. This means by default we'll make HTTP requests that are more aligned with what the browser would send by default.
- rename `mime_type` to `content_type` for consistency with the http header. Although I left `encoding` as-is.
- No longer ignore invalid form names, but to instead raise an exception
- Update nagios module to no longer need to explicitly override `nil` on most options
- Add a test for empty vars_form_data, as otherwise we would get form data metadata by default which was invalid
- Require users to explicitly specify filename, unless it can be inferred from the file object passed in
- Remove filename encoding, as some exploits put payloads here
- Update more modules to use the new api

### CODE REVIEW

File: spec/lib/rex/proto/http/client_spec.rb [hunk 1]
```diff
@@ -258,9 +258,24 @@ def excuse_needs_auth
...
     end
 
+    it 'should not include any form boundary metadata by default' do
+      request = cli.request_cgi({ })
+
+      expected = <<~EOF
+        POST / HTTP/1.1\r
+        Host: #{ip}\r
+        User-Agent: #{request.opts['agent']}\r
+        Content-Type: application/x-www-form-urlencoded\r
```
Comment (adfoster-r7) [inline spec/lib/rex/proto/http/client_spec.rb:268]: This is the existing behavior, I'd argue it's a bit weird - but I'm leaving this as-is for now :+1:

**Comment by sjanusz-r7** [comment]
Thanks!
The changes look good to me so far! I'm ready to merge this in once it is taken out of draft.
`filename = field_hash.fetch('filename') { get_filename(field_hash['data']) }` is something that I will use myself in the future 💯 
Also good spot on the issue of automatically encoding filenames. I didn't realise that could cause issues with payloads 😅 👍
